sndev

```diff
@@ -0,0 +1,162 @@
+#!/bin/sh
+
+docker__compose() {
+  CURRENT_UID=$(id -u) CURRENT_GID=$(id -g) command docker compose --env-file .env.sndev "$@"
+}
+
+docker__exec() {
+  command docker exec -i "$@"
+}
+
+docker__sn_lnd() {
+  docker__exec -u lnd sn_lnd lncli "$@"
+}
+
+docker__stacker_lnd() {
+  docker__exec -u lnd stacker_lnd lncli "$@"
+}
+
+docker__sn_lnd() {
+  t=$1
+  if [ "$t" = "-t" ]; then
+    shift
+  else
+    t=""
+  fi
+
+  docker__exec $t -u lnd sn_lnd lncli "$@"
+}
+
+docker__stacker_lnd() {
+  t=$1
+  if [ "$t" = "-t" ]; then
+    shift
+  else
+    t=""
+  fi
+
+  docker__exec $t -u lnd stacker_lnd lncli "$@"
+}
+
+sndev__start() {
+  if [ ! -x "$(command -v docker-compose)" ]; then
+    echo "docker compose is not installed"
+    echo "installation instructions are here: https://docs.docker.com/desktop/"
+    exit 0
+  fi
+
+  if ! [ -f .env.sndev ]; then
+    echo ".env.sndev does not exist."
+    echo "creating from .env.sample"
+    cp .env.sample .env.sndev
+  fi
+
+  docker__compose up --build
+}
+
+sndev__stop() {
+  docker__compose down
+}
+
+sndev__restart() {
+  docker__compose restart
+}
+
+sndev__status() {
+  shift
+  docker__compose ps --format 'table {{.Service}}\t{{.State}}\t{{.Status}}\t{{.Ports}}'
+}
+
+sndev__delete() {
+  echo "Deleting application"
+  docker__compose down --volumes --remove-orphans
+}
+
+sndev__fund() {
+  shift
+  docker__stacker_lnd -t payinvoice "$@"
+}
+
+sndev__withdraw() {
+  shift
+  docker__stacker_lnd addinvoice --amt "$@" | jq -r '.payment_request'
+}
+
+sndev__psql() {
+  docker__exec -t db psql -U sn stackernews
+}
+
+sndev__compose() {
+  shift
+  docker__compose "$@"
+}
+
+sndev__sn_lncli() {
+  shift
+  docker__sn_lnd -t "$@"
+}
+
+sndev__stacker_lncli() {
+  shift
+  docker__stacker_lnd -t "$@"
+}
+
+sndev__help() {
+    if [ $# -eq 3 ]; then
+        call "sndev__$1__$2__$3" "$@"
+    elif [ $# -eq 2 ]; then
+        call "sndev__$1__$2" "$@"
+    else
+        help="
+                            888
+                            888
+                            888
+      .d8888b  88888b.  .d88888  .d88b.  888  888
+     88K      888 '88b d88' 888 d8P  Y8b 888  888
+     'Y8888b. 888  888 888  888 88888888 Y88  88P
+          X88 888  888 Y88b 888 Y8b.      Y8bd8P
+      88888P' 888  888  'Y88888  'Y8888    Y88P
+
+manages a docker based stacker news development environment
+
+USAGE
+  $ sndev [COMMAND]
+
+COMMANDS
+  help            show help
+
+  env:
+    start         start env
+    stop          stop env
+    restart       restart env
+    status        status of env
+    delete        delete env
+
+  lnd:
+    fund          pay a bolt11 for funding
+    withdraw      create a bolt11 for withdrawal
```

Comment:
Arguments are missing. For `fund`, it was intuitive (simply pass bolt11) but for `withdraw` I got this error:

```
$ sndev withdraw
[lncli] flag needs an argument: -amt
jq: parse error: Invalid numeric literal at line 1, column 10
```

Then I tried `sndev withdraw -amt 1000`:

```
$ sndev withdraw -amt 1000
jq: parse error: Invalid numeric literal at line 1, column 10
[lncli] invalid value "-amt" for flag -amt: parse error
```

Then I looked at the code and realized it wanted a number:

```
$ sndev withdraw 1000
lnbcrt10u1pj7a6kapp5rxnlje73rmdrwvlglfyg78jvltwnss3lv23vq6ptccs005g2a09sdqqcqzzsxqyz5vqsp5kw77gyajrgyrtv9dreuhd7u9m73te8vfavcwllrh8zpgu20gsdqs9qyyssq86ae4ejj7t8t74jmu9n8tmx0s9xcm8gs6cvz7yk4tarww0fjnlks7w49vhalc6j7nmpqdegq9l9ncvjely7edesp3lvxaf78es9w4ccqsntl0p
```